### ISSUE TITLE
Clarification on "sparse token prediction" or "sparse output prediction"

### ISSUE THREAD

**Issue by mirandrom** [IntelLabs/academic-budget-bert#13]
First off, thanks for the great paper and making your code publicly available, it's been really valuable.

I was hoping to clarify what you meant by "sparse token prediction" (in Section 3.1, under Software) and "sparse output prediction" (in Table 2). For both instances, you cite the original RoBERTa paper, however I can't find any mention of sparsity in their paper.

In your model args, you define sparse mask prediction as predicting only masked tokens: https://github.com/IntelLabs/academic-budget-bert/blob/ea000838156e3be251699ad6a3c8b1339c76e987/pretraining/args/model_args.py#L105

Is this what is meant in the paper? If so, what is the time saved reported in Table 2 relative to?
If not, what do you mean by sparse token/output prediction?

Thank you!

**Comment by peteriz** [comment]
Thank you, I appreciate your feedback

> Is this what is meant in the paper? If so, what is the time saved reported in Table 2 relative to? If not, what do you mean by sparse token/output prediction?

Correct; Line 1 of table 2 shows how much slow-down will occur to the training throughput or, alternatively, how much training (iterations) we would lose by not using that feature in given time budget.
It wasn't mentioned in RoBERTa's paper however its available in the official implementation [here](https://github.com/pytorch/fairseq/blob/main/fairseq/models/roberta/model.py#L380).

**Comment by mirandrom** [comment]
Thanks, that clears everything up!
